add/add.js

```diff
@@ -0,0 +1,32 @@
+import { createParticipant, getWorkshops, checkAuth } from '../fetch-utils.js';
+
+const addForm = document.getElementById('add-form');
+const selectWorkshop = document.getElementById('workshop-select');
+const cancelButton = document.getElementById('cancel');
+
+checkAuth();
+
+addForm.addEventListener('submit', async (e) => {
+    e.preventDefault();
+    const formData = new FormData(addForm);
+    const name = formData.get('name');
+    const workshopID = formData.get('workshop-id');
+    await createParticipant(name, workshopID);
+    addForm.reset();
+    window.location.replace('/workshops');
+
+});
+
+window.addEventListener('load', async () => {
+    const workshops = await getWorkshops();
+    for (let workshop of workshops) {
+        const option = document.createElement('option');
+        option.value = workshop.id;
+        option.textContent = workshop.name;
+        selectWorkshop.append(option);
+    }
+});
+
+cancelButton.addEventListener('click', () => {
+    window.location.replace('/workshops');
+});
```

Comment:
No biggie here, but when you see these red circles with a line like on 32 github expects a new line at the bottom of files. If you use prettier it can be automated and you'll never have to think about that. If you are curious about that reach out and I'd be happy to help you set it up!